### ISSUE TITLE
error when using pictures with others apps

### ISSUE THREAD

**Issue by FilLupin** [k3b/APhotoManager#56]
Hi, 
after deleting a picture with aPhotoManager,  I should be able to use normally gallery and not see this picture anymore.
Same for camera which I should be able to use.

However, when using gallery, the thumb of the deleted picture is still present (not visible, this is the "normal" behavior, but the thumb still take place between other pictures) and when selecting it to watch it, it make gallery app stop.
When launching camera after deleting the last photo I took with camera, camera stops immediatly. perhaps because it failed to print the last picture.

My system is a Samsung S3 with Android 4.2.2 with 3.0.64 kernel version
I have this issue with [A Photo Manager Version 19](https://github.com/k3b/APhotoManager/releases/tag/v0.5.2.160726) (i.e. 0.5.2.160726) and formerly [A Photo Manager Version 17](https://github.com/k3b/APhotoManager/releases/tag/v0.5.0.160629) (i.e. 0.5.0.160629)

* Affected Module:
	The real bug does not affect  A Photo Manager app but other app wich interact with it, i.e. gallery and camera.

**Comment by k3b** [comment]
I am not shure if i understand you correctly what you mean by "the thumb is still present".

Do yu mean this

* start stock gallery showing "pic-a.jpg" in gallery mode
* start aPhotoManager
* delete "pic-a.jpg" with aPhotoManager
* android: switch back to stock gallery still showing "pic-a.jpg" in gallery mode
* stock gallery: select "pic-a.jpg" to view it => stock gallery crashes

(i havent tried yet if this happens on my devices)

Does the same problem exisits if you delete a photo with other non-stock gallery apps or a filemanager?

Do you think that the problem can be solved if i manually delete the android-stock thumb-files that belong to the deleted images?

**Comment by FilLupin** [comment]
Hi,
yes, the steps are correct except that there seems to be no need to start gallery at first step (app can be stopped).
While deleting a picture with filemanager in /sdcard/DCIM/Camera it still appears into both gallery and aPhotoManager. 
Do you have any idea of the path where I have to delete this picture ?

**Comment by k3b** [comment]
You haven-t answered (3) yet.

To explain why the delete image is still visible:

the jpg-image exists 

* as a file in the folder /sdcard/DCIM/Camera/ and 
* as a database entry in the android-media-db-contentenprovider with 
* a precomputed miniversion of the picture (thumb)

Delting the file alone with a filemanager does not immediatly delete the item in media-db and it-s thumb. this is done by android-s media scanner after some time.  

media-db+thumb are used to find/sort and to display in a galleryview. if these are not deleted they are still visible.

aPhotoManager and most other gallery apps delete file mediadb and thumb together.

however i assume that stock-gallery-app does not know about aPhotoManger-s delete and i assume that it does not listen to the media-db-channel that something has changed.

I am not shure if i can do anythink to solve the problem you reported.

**Comment by FilLupin** [comment]
Sorry for the delay, I did not saw your answer. 
After restarting, the gallery still give access to the buggy picture and crashes when clicked.
However, I saw that the error message is about the camera which crashes when I tried to access the deleted picture, and not the gallery.
I do not know the gallery application enough to conclude anything.

Whatever, if aPhotoManager deletes the picture and both the thumb and the db entry, then I imagine there is perhaps some cache into gallery. Your answer made me think to erase the cache of gallery and camera, but it does not change anything. Even when I force to stop both camera and gallery, it does not change anything... 
I will try to find the gallery dev team to report this issue to them.
